.github/workflows/build.yml

```diff
@@ -18,7 +18,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [
-          macOS-latest,
+          macos-14,
```

Comment:
This uses the newer ARM based macOS runner and makes `composeA11y`, `customFontsInCode`, and `customFontsInXml`. The 3 tests would still fail on Intel macOS, it seems that that variant of layoutlib has a small but bigh enough rendering difference compared to macOS ARM, Linux and Windows. 

I've also did a test with the latest tagged Iguana layoutlib by publishing it myself and with the same changes as this PR (except for the last commit [3b2c9c4](https://github.com/cashapp/paparazzi/pull/1228/commits/3b2c9c4da4d28cd172a796991f9c3d2152513d13)) Iguana also works: https://github.com/freeletics/paparazzi/actions/runs/7731866563